components/builder-api/src/server/resources/jobs.rs

```diff
@@ -432,7 +432,11 @@ fn do_get_job(req: &HttpRequest<AppState>, job_id: u64) -> Result<String> {
 
             if job.get_package_ident().fully_qualified() {
                 let builder_package_ident = BuilderPackageIdent(job.get_package_ident().into());
-                let channels = channels_for_package_ident(req, &builder_package_ident, &*conn)?;
+                let channels =
+                    channels_for_package_ident(req,
+                                               &builder_package_ident,
+                                               PackageTarget::from_str(job.get_target()).unwrap(),
```

Comment:
Is it possible for this `unwrap` to fail or is `job.get_target()` guaranteed to return a valid target?

Reply:
Good q - all jobs must have a valid target to be entered in the db, so we should be good with this unwrap

Reply:
The original failure in the release process that led me to this was an `unwrap` that I thought could never fail (an arg that was required by clap, see https://github.com/habitat-sh/habitat/pull/6363). It would've been *much* faster to locate the problem if that code were an `expect` originally, so I'm of the mind to always use expect at this point.